### PR TITLE
Created the DBApi

### PR DESCRIPTION
**Pull request by kojapp**
## **Description**
Created a database api that will handle the requests to the database. For now, the available actions are GET, POST, PUT, DELETE. Remarks: the bash script to start the database api will not be source controlled as it contains the private URI that connects to Heroku, instead it will be provided separately.

## **Testing**
Tested each route with PostMan, making sure that the routes effectively create, read, update and delete the rows of the database.

### CODE REVIEW

File: routes/database-api.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        url: `http://${domain}:${targetPORT}/${database}/tables`,
+        method: 'GET'
+    }
+    request(options, (err, response, body) => {
+        if (err) throw err;
+        else res.send(body);
+    });
+});
+
+router.get('/:database/:table', (req, res) => {
```
Comment (jbennyhuang) [inline routes/database-api.js:22]: get items within that table?

File: routes/database-api.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
+const express = require('express');
+const request = require('request');
+const router = express.Router();
+const domain = 'localhost';
+const targetPORT = 9191;
+
+// these routes send requests to the specified :database
+// refer to the DBApi for information on the various routes
+
+router.get('/:database/tables', (req, res) => {
```
Comment (jbennyhuang) [inline routes/database-api.js:10]: get list of tables?

File: routes/database-api.js (hunk 1 above)
Comment (kojapp) [reply routes/database-api.js:22]: yes

File: routes/database-api.js (hunk 2 above)
Comment (kojapp) [reply routes/database-api.js:10]: yes
refer to the DBApi for more info

File: services/DBApi/databases/libs/query-format.js [hunk 3]
```diff
@@ -0,0 +1,139 @@
+/**
+ * @function getQueryFormat(...) transforms a key-value dictionary into a valid SQL Query for GET requests
+ * 
+ * @param {JSON} queryParams    : a dictionary where keys are column names and values are entries of the columns
```
Comment (jbennyhuang) [inline services/DBApi/databases/libs/query-format.js]: What is the format of queryParams in URL form
e.g. localhost:3000/dbname/tablename?[WHAT GOES HERE]

Comment (kojapp) [reply services/DBApi/databases/libs/query-format.js]: localhost:3000/dbname/tablename?username=someuser&password=somepw

=> queryParams will be { username: "someuser", password: "somepw" }

Comment (jbennyhuang) [reply services/DBApi/databases/libs/query-format.js]: How about for POST, PUT and DELETE? I get some errors with the above in postman

Comment (kojapp) [reply services/DBApi/databases/libs/query-format.js]: Under Body, select raw and JSON (application/json)

For POST and DELETE, the keys are columns and values are entries, e.g.
{ "username": "someuser", "password": "somepw" }

For PUT, it is a '2-dimensional' json, you need to specify what are the updates and what are the conditions, e.g.
{ "updates": { "password": "newpw", "email": "newemail" }
  "conditions": { "username": "onlyupdatethisuser" } }

**Pull request merged**